### PR TITLE
JEN-960 add '--sanitize' option in case of 'UBSAN/ASAN' builds

### PR DESCRIPTION
**Pull request by hors**
* add '--sanitize' option in case of '-WITH_UBSAN/-WITH_ASAN' builds
    * fix 'UBSAN' builds for rhel8

### CODE REVIEW

**Comment by hors** [comment]
https://ps3.cd.percona.com/job/percona-server-8.0-pipeline/2060/

**Review by george-lorch — APPROVED** [review]
+1 pending Yura review

**Comment by hors** [comment]
https://ps3.cd.percona.com/job/percona-server-8.0-pipeline/2064 - UBSan fix for rhel8

File: local/test-binary [hunk 1]
```diff
@@ -90,15 +90,19 @@ if [[ "${ANALYZER_OPTS}" == *WITH_VALGRIND=ON* ]]; then
     PARALLEL=$((PARALLEL/3))
   fi
 fi
-#
+
+if [[ "${ANALYZER_OPTS}" == *WITH_ASAN=ON* ]] || [[ "${ANALYZER_OPTS}" == *WITH_UBSAN=ON* ]]; then
```
Comment (percona-ysorokin) [inline local/test-binary]: `${ANALYZER_OPTS}` can also be `-DWITH_MSAN=ON` (even though we do not test this currently).
I am not also sure about `WITH_ASAN_SCOPE=ON: as far as I remember `WITH_ASAN_SCOPE=ON` always comes along with `WITH_ASAN=ON` - please double check.

**Review by percona-ysorokin — CHANGES_REQUESTED** [review]

File: local/test-binary (hunk 1 above)
Comment (hors) [reply local/test-binary]: you are right, WITH_ASAN_SCOPE should be used with WITH_ASAN

from documentation:
-DWITH_ASAN_SCOPE=bool
Whether to enable the AddressSanitizer -fsanitize-address-use-after-scope Clang flag for use-after-scope detection. The default is off. To use this option, -DWITH_ASAN must also be enabled.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**